mne/gui/_file_traits.py

```diff
@@ -288,53 +305,57 @@ def _get_n_omitted(self):
 
     @cached_property
     def _get__info(self):
-        if self.file:
+        if not self.file:
+            return
+        elif self.file.endswith(('.fif', '.fif.gz')):
             info = None
             fid, tree, _ = fiff_open(self.file)
             fid.close()
             if len(dir_tree_find(tree, FIFF.FIFFB_MEAS_INFO)) > 0:
                 info = read_info(self.file, verbose=False)
             elif len(dir_tree_find(tree, FIFF.FIFFB_ISOTRAK)) > 0:
                 info = read_dig_fif(fname=self.file)
+        else:
+            info = read_raw(self.file).info
+
+        if isinstance(info, DigMontage):
+            dig = info.dig
+            info = _empty_info(1)
+            info['dig'] = dig
+        elif info is None or info['dig'] is None:
+            error(None, "The selected file does not contain digitization "
+                  "information. Please select a different file.",
+                  "Error Reading Digitization File")
+            self.reset_traits(['file'])
+            return
+        else:
+            # check that all fiducial points are present
+            has_point = {FIFF.FIFFV_POINT_LPA: False,
+                         FIFF.FIFFV_POINT_NASION: False,
+                         FIFF.FIFFV_POINT_RPA: False}
+            for d in info['dig']:
+                if d['kind'] == FIFF.FIFFV_POINT_CARDINAL:
+                    has_point[d['ident']] = True
+            if not all(has_point.values()):
```

Comment:
It seems like this whole section could be simplified, DRY'd, and have fewer missed coverage lines with something like:
```
missing = [key for key in ('LPA', 'Nasion', 'RPA') if not 
           has_point[getattr(FIFF, f'FIFFV_POINT_{key.upper()}']
if missing:
    points = _fiducial_coords(info['dig'])
    if len(points == 3):
        _append_fiducials(info['dig'], *points.T)
    else:
        error(...)
        self.reset_traits(['file'])
        return
```